biz.aQute.launcher/src/aQute/launcher/Launcher.java

```diff
@@ -471,15 +480,39 @@ private int activate() throws Exception {
 		return result;
 	}
 
-	private boolean isImmediate(BundleActivator activator) {
+	private boolean isAfterFrameworkStart(BundleActivator activator) {
 		try {
 			Field f = activator.getClass()
 				.getField("IMMEDIATE");
+			if (f.getType()
+				.equals(Boolean.class)
+				|| f.getType()
+					.equals(boolean.class)) {
+				return false;
+			} else if (f.getType()
+				.equals(String.class)) {
+				return "AFTER_FRAMEWORK_START".equals(f.get(activator));
+			}
```

Comment:
Could you extract the commonality with the next method? It is almost identical. Just make a method that returns the value as an Object